flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/table/catalog/hive/HiveCatalogGenericMetadataTest.java

```diff
@@ -288,6 +293,45 @@ public void testTableSchemaCompatibility() throws Exception {
         }
     }
 
+    @Test
+    public void testFunctionCompatibility() throws Exception {
+        catalog.createDatabase(db1, createDb(), false);
+        catalog.createFunction(path1, new CatalogFunctionImpl("flink:class.name"), false);
```

Comment:
Is the `flink:` prefix valid? Will Flink framework add the prefix?

Reply:
Yes. Prior to this PR, we add the `flink:` prefix for generic functions created in HiveCatalog. This test is to make sure such functions can still be properly retrieved after this change.

Reply:
IIUC, the `flink:` prefix is added by HiveCatalog in `instantiateHiveFunction`. Therefore, the class name in `CatalogFunction` shouldn't contain `flink:` prefix, because it is passed by the Flink framework, it will never pass a "flink:class.name" to a catalog.

However, it seems that we can't get the metastore client here for writing a specific function into it... Maybe we can add a comment above this line to explain why we use a `flink:` prefixed class name here.

Reply:
I see your point. Actually we can directly call metastore client to create the function and I agree that'll make the test easier to understand.